The War Against Drugs Pt.1: The Cocanium Threat

Wernerses opened the pull request:
## The War Against Drugs Pt.1: The Cocanium Threat

### My dear crewmates, what is the biggest threat to our space company?

Is it the violent abuse of security against crew?
<img width="308" height="195" alt="LfGj9u" src="https://github.com/user-attachments/assets/d9cdedae-4385-405b-a055-203cd895a28f" />

Is it the threat of atmosians constantly plasmaflooding our homes?
<img width="308" height="195" alt="bSRiJn" src="https://github.com/user-attachments/assets/c8a08926-656b-4434-b10e-f9c487a01bfd" />

Is it the menace of botany and their diabolical plants?

<img width="308" height="195" alt="ZKh1az" src="https://github.com/user-attachments/assets/f36d9728-7963-4559-a0a0-a948827a88fb" />

### THEN WAHT IS IT DR. SHOES?
<img width="577" height="418" alt="MR1QHa" src="https://github.com/user-attachments/assets/ef36ea03-cc8c-4a2d-bd6e-60e873078471" />

## Chapter 1: The Problem, Your Consumption

https://github.com/user-attachments/assets/ec21718b-1352-4173-abf5-874c547eaf73

In a grave event, known as "The Whiteout", crew, through the elaborate and malicious efforts of the Department of Cargo, and of Service unionized to create a terrorist against our very ~~nation~~ company!!!

This cruel duo of botanical science and maniacal unrelenting labour greedyness caused a space anomaly that caused the spacetime continuum of that station to be turned into a flat pancake.

### I Say: No More My Fellow Crewmates

## Chapter 2: The Solution, Your Compartamentalization

Through the efforts of a loyal member of our company, who shall remain unnamed in order to protect his integrity and to avoid the vile accusations of being called a quote-unquote "SCAB", we have reached a scientific discovery: The Compartamentalization of Cocaine

<img width="340" height="225" alt="jXave3" src="https://github.com/user-attachments/assets/90590572-99e1-4cd1-bc1d-87618b1e0784" />
<img width="350" height="225" alt="CVTBfO" src="https://github.com/user-attachments/assets/200b0395-45b3-427d-beee-0d28d671dad1" />

Now, Cocaine stacks with itself, preventing space time anomalies, and more:

### It is now considered A MATERIAL SHEET

But do not fret, my fellow crewmates, we are yet far away from the possibilities of furniture and other garments made out of this vile material. For now, this is only possible for carving blocks and some other... mysteriours chemical interactions.

This is part one, of a series of changes yet to come in order to bring about the greatest fear of all drug traffickers and illegal actors:

### Governamental Regularization!!

## Chapter 3: Nixon, The White

### WHAT? YOU THOUGHT I WAS GOING TO MAKE COCAINE ILLEGAL? 
<img width="666" height="360" alt="IbGp29" src="https://github.com/user-attachments/assets/fa0b7e6e-1ca7-4c10-ae79-05312e0ae266" />

My dear colleagues, my beloved co-workers, cocaine is the grease of our economy, the gears of our consumption, the oil in our car, it drives us for more, it makes look towards a better future!

With this change, Cargo and Botany can now better work together with no longer fear of staring into a floor and suddenly dying inside unable to process all the separate individual cocaine piles.

More importantly, crew can now enjoy more manageable piles of cocaine, and, if by desire, CentCom officers can even make cocaine piles **not** contain any reagents inside (through variable changes) making what i and my fellow associates call

### " Family Friendly Cocaine!! "

## Chapter 4: A Colorful Future

![7yYDQN](https://github.com/user-attachments/assets/164cd867-7dce-4c56-9770-f1a91ecfee68)

With this change my fellow crewmates, this is our future:
* A cargo, more equipped to send even more drugs to even more happier families
* A botany, that no longers break reality by just doing work 
* A station, unified and richer, through the means of drugs
* A brighter, new future for our Monkestation

## Why It's Good For The Game
* Helps avoid the clogging of botany with excessive individual piles of cocaine (like of the shown video)
* Allows easier transportation of said piles
* Creates a way to have objects made out of cocaine (yet to made more in depth)

## Changelog 

:cl: A Joyful contribution from Ancient_Engineer and Dr. Shoes
qol: Made so cocaine piles now stack rather than working as individual objects
add: Cocaine piles are now considered material sheets, which can possibly have funny interactions (more possibilities to come???)
/:cl:

Wernerses committed:
The Cocainum Threat

Wernerses committed:
The Start of Breaking Reality of Cocaine

Wernerses committed:
Cocainum Deletion Plus Substitution

Wernerses committed:
Removing Cocaine Inhand Sprite Code

Maybe this helps?

ancient-engineer commented on code/game/objects/items/stacks/sheets/sheet_types.dm:
```diff
@@ -893,3 +893,63 @@ new /datum/stack_recipe("paper frame door", /obj/structure/mineral_door/paperfra
+	to_chat(user, span_notice("You finish snorting [src]."))
+	if(reagents.total_volume)
+		reagents.trans_to(user, reagents.total_volume, transfered_by = user, methods = INGEST)
+	qdel(src)
```
```suggestion
	use(1)
```

ancient-engineer commented on code/game/objects/items/stacks/sheets/sheet_types.dm:
```diff
@@ -893,3 +893,63 @@ new /datum/stack_recipe("paper frame door", /obj/structure/mineral_door/paperfra
+	resistance_flags = FLAMMABLE
+	novariants = TRUE
+	grind_results = list(/datum/reagent/drug/cocaine = 10)
+
```
```suggestion
/obj/item/stack/sheet/cocaine/Initialize(mapload)
	. = ..()
	for(var/reagent in grind_results)
		var/amount += grind_results[reagent]
	create_reagents(amount)
	reagents.add_reagent_list(grind_results)
```

Thermosilver reviewed (commented):
(:

Wernerses committed:
Update code/game/objects/items/stacks/sheets/sheet_types.dm

Co-authored-by: ancient-engineer <103686984+ancient-engineer@users.noreply.github.com>

Wernerses committed:
Update code/game/objects/items/stacks/sheets/sheet_types.dm

Co-authored-by: ancient-engineer <103686984+ancient-engineer@users.noreply.github.com>

Wernerses committed:
Final Changes Before Setting it up fo Review

* Formated the code as instructed by the amazing Ancient Engineer

* Gave another stack possibility of 100, may this prevent cocaine instability

Thermosilver reviewed (commented):
Small fix

Wernerses committed:
Update code/game/objects/items/stacks/sheets/sheet_types.dm

Co-authored-by: ancient-engineer <103686984+ancient-engineer@users.noreply.github.com>

Thermosilver reviewed (commented):
This will probably work better

Wernerses committed:
Update code/game/objects/items/stacks/sheets/sheet_types.dm

Co-authored-by: ancient-engineer <103686984+ancient-engineer@users.noreply.github.com>

Wernerses committed:
Update code/game/objects/items/stacks/sheets/sheet_types.dm

Co-authored-by: ancient-engineer <103686984+ancient-engineer@users.noreply.github.com>

Wernerses committed:
Update code/game/objects/items/stacks/sheets/sheet_types.dm

Co-authored-by: ancient-engineer <103686984+ancient-engineer@users.noreply.github.com>

TheColorCyan commented on code/datums/materials/basemats.dm:
```diff
@@ -589,3 +589,17 @@ monkestation end */
+	greyscale_colors = "#ffffffec"
+	categories = list(MAT_CATEGORY_RIGID = TRUE, MAT_CATEGORY_ITEM_MATERIAL=TRUE)
+	sheet_type = /obj/item/stack/sheet/cocaine
+	value_per_unit = 20
```
value_per_unit and beauty_modifier should use defines (SHEET_MATERIAL_AMOUNT)

Wernerses replied:
oH shoot, taking a look at it rn

Wernerses replied:
Solved! 

Thanks for pointing it out. Cargo would've taken my eyes off for this one.

Nice catch.

Wernerses committed:
Merge branch 'Monkestation:master' into The-Cocainum-Threat

Wernerses committed:
Updated The Cocaine sheets to consider SHEET_MATERIAL_AMOUNT

Man this could've made these WORTHLESS

Thanks TheColorCyan for pointing it out before cargo took my eyes off.

Wernerses committed:
Made the Sheets EVEN UGLIER

>Sees cocaine
>Depression

ThePooba committed:
Merge branch 'master' into The-Cocainum-Threat

Wernerses committed:
Merge branch 'master' into The-Cocainum-Threat

Wernerses committed:
Merge branch 'master' into The-Cocainum-Threat

Wernerses committed:
Fixed the extra line, and removed the 100 stack

Unsure if the 100 one is really that necessary.

Wernerses commented on code/game/objects/items/stacks/sheets/sheet_types.dm:
```diff
@@ -893,3 +893,66 @@ new /datum/stack_recipe("paper frame door", /obj/structure/mineral_door/paperfra
+	amount = 20
+/obj/item/stack/sheet/cocaine/fifty
+	amount = 50
+/obj/item/stack/sheet/cocaine/fifty
```
Nevermind, i was looking at the changed files wrong.

Anyways, thank you so much Throw.
